tests/v2/integration/catalogv2/system_charts_version_test.go

```diff
@@ -0,0 +1,272 @@
+	defer w.resetSettings()
+
+	const minVersion = "2.0.3+up0.3.3"
+	w.Require().NoError(w.uninstallApp("cattle-system", "rancher-webhook"))
+	w.Require().NoError(w.updateSetting("rancher-webhook-min-version", minVersion))
+
+	watcher, err := w.catalogClient.Apps("cattle-system").Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + "rancher-webhook",
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	w.Require().NoError(err)
+
+	err = wait.WatchWait(watcher, func(event watch.Event) (ready bool, err error) {
+		if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error installing the rancher-webhook chart")
+		} else if event.Type == watch.Added {
+			return true, nil
+		}
+		return false, nil
+	})
+	w.Require().NoError(err)
+
+	// Allow the new release to fully deploy. Otherwise, the client won't find it among current releases.
+	var newRelease *release.Release
+	err = kwait.Poll(10*time.Second, time.Minute, func() (done bool, err error) {
+		newRelease, err = w.fetchRelease("cattle-system", "rancher-webhook")
+		if err != nil {
+			return false, nil
+		}
+		if v := newRelease.Chart.Metadata.Version; v != w.latestWebhookVersion {
+			w.T().Logf("%s version %s does not yet match expected %s", newRelease.Chart.Name(), v, w.latestWebhookVersion)
+			return false, nil
+		}
+		return true, nil
+	})
+	w.Require().NoError(err)
+}
+
+func (w *SystemChartsVersionSuite) TestInstallFleet() {
+	defer w.resetSettings()
+
+	w.Require().NoError(w.uninstallApp("cattle-fleet-system", "fleet"))
+
+	const minVersion = "102.0.0+up0.6.0"
+	w.Require().NoError(w.updateSetting("fleet-min-version", minVersion))
+
+	watcher, err := w.catalogClient.Apps("cattle-fleet-system").Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + "fleet",
+		TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+	})
+	w.Require().NoError(err)
+
+	err = wait.WatchWait(watcher, func(event watch.Event) (ready bool, err error) {
+		if event.Type == watch.Error {
+			return false, fmt.Errorf("there was an error installing the fleet chart")
+		} else if event.Type == watch.Added {
+			return true, nil
+		}
+		return false, nil
+	})
+	w.Require().NoError(err)
+
+	// Allow the new release to fully deploy. Otherwise, the client won't find it among current releases.
+	var newRelease *release.Release
+	err = kwait.Poll(10*time.Second, 2*time.Minute, func() (done bool, err error) {
+		newRelease, err = w.fetchRelease("cattle-fleet-system", "fleet")
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	w.Require().NoError(err)
+
+	latest, err := w.catalogClient.GetLatestChartVersion("fleet")
+	w.Require().NoError(err)
+
+	// Ensure Rancher deployed the latest version when the minimum version is below the latest.
+	w.Assert().Equal(newRelease.Chart.Metadata.Version, latest)
+}
+
+func (w *SystemChartsVersionSuite) uninstallApp(namespace, chartName string) error {
```

Comment:
uninstallApp, getReleases, fetchReleases. Seem like good candidates for extensions, if we are to reuse this functionality for future chart tests.

Reply:
Agreed! I'd be glad to do it, but I might not have time to factor this out in this PR - this issue needs to be complete by EOD.